frontend/src/components/RoomList/RoomList.js

```diff
@@ -1,19 +1,38 @@
-import React from 'react';
+import React, { useState, useMemo } from 'react';
 import PropTypes from 'prop-types';
 import RoomListItem from './RoomListItem/RoomListItem';
-import { StyledRoomList } from './RoomList_styles';
+import RoomFilterOptions from './RoomFilterOptions/RoomFilterOptions';
+import RoomMetadataDTO from '../../services/parsing/RoomMetadataDTO';
+import { StyledRoomList, FilteredRoomList } from './RoomList_styles';
 
 const RoomList = ( { roomsData } ) => {
 
-   const rooms = roomsData.map( ( ( room )=> (
-      <RoomListItem key={ room.id } roomData={ room } />
-   ) ) );
+   const [ filters, setFilters ] = useState( {} );
+
+   const rooms = useMemo( () => {
+      return roomsData
+         .map( ( room ) => ( { room, metadata: RoomMetadataDTO.from( room.summary, room.description ) } ) )
+         .filter( ( { metadata } ) => filters.hasProjector ?
+            metadata.hasProjector : true )
+         .filter( ( { metadata } ) => filters.hasWhiteboard ?
+            metadata.hasWhiteboard : true )
+         .filter( ( { metadata } ) => filters.seatsNo ?
+            metadata.seatsNo >= filters.seatsNo : true )
+         .filter( ( { metadata } ) => filters.name ?
+            metadata.name.toLowerCase().includes( filters.name.toLowerCase() ) : true )
```

Comment:
This string of `filter` functions could be easily extracted as separate `filterRooms` function for code clarity.

Reply:
Fixed in #60 .